api/v1alpha1/httproute_types.go

```diff
@@ -72,15 +72,25 @@ type HTTPRouteHost struct {
 
 // HTTPRouteRule is the configuration for a given path.
 type HTTPRouteRule struct {
-	// Match defines which requests match this path.
+	// Match defines a schema for matching an HTTP request.
+	//
+	// Support: core
+	//
 	// +optional
-	Match *HTTPRouteMatch `json:"match" protobuf:"bytes,1,opt,name=match"`
-	// Filter defines what filters are applied to the request.
+	Match *HTTPRequestMatch `json:"match" protobuf:"bytes,1,opt,name=match"`
+	// Filter defines a schema for applying filters to a matched HTTP request.
```

Comment:
The doc comment for filter in the target is more specific and helpful than this one:

> // Filter defines filters to apply to the request before forwarding it to
	// a targetRef.


Can we also add the semantic meaning of the filter in this godoc?
Essentially, saying that the filters MUST be executed when this route is selected for a request.